src/Compilers/Core/MSBuildTask/Shared/ManagedCompiler.cs

```diff
@@ -922,6 +928,20 @@ bool resultFromHostObjectSetOperation
             }
         }
 
+        internal string InitializeHostObjectSupportForNewSwitches(ITaskHost hostObject, ref string param)
+        {
+            var compilerOptionsHostObject = hostObject as ICompilerOptionsHostObject;
+
+            if (compilerOptionsHostObject != null)
+            {
+                var commandLineBuilder = new CommandLineBuilderExtension();
+                AddResponseFileCommandsForSwitchesSinceInitialReleaseThatAreNeededByTheIDE(commandLineBuilder);
+                CheckHostObjectSupport(param = "CompilerOptions", compilerOptionsHostObject.SetCompilerOptions(commandLineBuilder.ToString()));
```

Comment:
uh... why're assigning to the ref inline as you call an invocation expression?  can we not do that?